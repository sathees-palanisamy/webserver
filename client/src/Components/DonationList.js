import { useSelector, useDispatch } from "react-redux";
import * as actions from "../Store/actions";
import { useState, useEffect } from "react";
import Nav from "./Nav";
import DonationPresentation from './DonationPresentation';

function DonationList(props) {
  const errorDetail = useSelector((state) => state.donation.listStatus);
  const reduxDonationList = useSelector((state) => state.donation.donationList);
  
  const dispatch = useDispatch();


  let donationListDetails;

  const deleteDonation = (id) => {
     dispatch(actions.deleteInPorgress());
     dispatch(actions.deleteDonationItem(id));
     dispatch(actions.getDonationList());
  }

  if (reduxDonationList.length > 0) {
    donationListDetails = reduxDonationList.map((indiDonation) => {
      return <DonationPresentation 
              id={indiDonation.id}
              amount={parseInt(indiDonation.amount)}
              userid={indiDonation.userid}
              tip={indiDonation.tip}
              deleteDonation={deleteDonation}
              key={indiDonation.id}
              
              />
    });
  }

  useEffect(() => {
    dispatch(actions.listInPorgress());
    dispatch(actions.getDonationList());
  }, []);

  return (
    <div className='container' data-test="donation-list">
      <h1 data-test="donation-list-heading">Donation List</h1>

      <div data-test="donation-details">{donationListDetails}</div>

      <span>{errorDetail}</span>
    </div>
  );
}

export default DonationList;
