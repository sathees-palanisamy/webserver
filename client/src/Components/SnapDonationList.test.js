import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import { storeFactory } from '../test/testutil';
import DonationList from './DonationList';

// activate global mock to make sure getSecretWord doesn't make network call
jest.mock("../Store/actions");

const store = storeFactory({});

it('renders no donation', () => {
  const tree = renderer.create(<Provider store={store}><DonationList  /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

const defaultState = {
  donation : {
  currentUser: "",
  currentDonationAmount: 0,
  currentTipAmount: 0,
  donationList: [
    {
      "id": 1634585325626,
      "userid": "1",
      "amount": "0",
      "tip": 10
  },
  {
      "id": 1639962674693,
      "userid": "1",
      "amount": "400",
      "tip": 4
  },
  {
      "id": 1642205197746,
      "userid": "1",
      "amount": "140",
      "tip": 14
  },
  {
      "id": 1,
      "userid": "1",
      "amount": "700",
      "tip": 7
  }
],
  insertStatus: "",
  listStatus: "",
}
}

const store2 = storeFactory({...defaultState});

it('renders list of donation', () => {
  const tree = renderer.create(<Provider store={store2}><DonationList  /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});