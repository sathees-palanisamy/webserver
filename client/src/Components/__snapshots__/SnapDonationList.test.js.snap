// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of donation 1`] = `
<div
  className="container"
  data-test="donation-list"
>
  <h1
    data-test="donation-list-heading"
  >
    Donation List
  </h1>
  <div
    data-test="donation-details"
  >
    <div
      className="Card"
      data-test="donation-presentation"
    >
      <div
        className="Card--body"
        data-test="Card--body"
      >
        <h4
          className="Card--body-title"
          data-test="Card--body-title"
        >
          Donation Amount:
          400
        </h4>
        <p
          className="Card--body-tip"
          data-test="Card--body-tip"
        >
          Tip :
          4
        </p>
        <p
          className="Card--body-userid"
          data-test="Card--body-userid"
        >
          User ID :
          1
        </p>
      </div>
      <button
        className="Card__button"
        data-test="Card__button"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="Card"
      data-test="donation-presentation"
    >
      <div
        className="Card--body"
        data-test="Card--body"
      >
        <h4
          className="Card--body-title"
          data-test="Card--body-title"
        >
          Donation Amount:
          140
        </h4>
        <p
          className="Card--body-tip"
          data-test="Card--body-tip"
        >
          Tip :
          14
        </p>
        <p
          className="Card--body-userid"
          data-test="Card--body-userid"
        >
          User ID :
          1
        </p>
      </div>
      <button
        className="Card__button"
        data-test="Card__button"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="Card"
      data-test="donation-presentation"
    >
      <div
        className="Card--body"
        data-test="Card--body"
      >
        <h4
          className="Card--body-title"
          data-test="Card--body-title"
        >
          Donation Amount:
          700
        </h4>
        <p
          className="Card--body-tip"
          data-test="Card--body-tip"
        >
          Tip :
          7
        </p>
        <p
          className="Card--body-userid"
          data-test="Card--body-userid"
        >
          User ID :
          1
        </p>
      </div>
      <button
        className="Card__button"
        data-test="Card__button"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
  <span>
    success
  </span>
</div>
`;

exports[`renders no donation 1`] = `
<div
  className="container"
  data-test="donation-list"
>
  <h1
    data-test="donation-list-heading"
  >
    Donation List
  </h1>
  <div
    data-test="donation-details"
  />
  <span>
    
  </span>
</div>
`;
